packages/playwright-core/src/server/injected/injectedScript.ts

```diff
@@ -1072,10 +1070,8 @@ export class InjectedScript {
       for (let i = 0; i < children.length; i++)
         onlyText = onlyText && children[i].nodeType === Node.TEXT_NODE;
     }
-    let text = onlyText ? (element.textContent || '') : (children.length ? '\u2026' : '');
-    if (text.length > 50)
-      text = text.substring(0, 49) + '\u2026';
```

Comment:
I wonder whether we have more `substring()` calls that produce bad UTF16? Can we somehow audit our usage?

Reply:
I did a quick audit by grepping for `.substring(0`, thats how I found the one in `page.ts`.

Reply:
I found this one as well: https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/server/injected/selectorGenerator.ts#L279